### PR TITLE
Feature/parameter dependency rules

### PR DESCRIPTION
**Pull request by sudarshan0212**
Thank you for your contribution.
Before submitting this PR, please make sure:
- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Help Libray document is updated accordingly for Feature changes.

### CODE REVIEW

File: Ginger/Ginger/UserControlsLib/MultiSelectComboBox.xaml
[the remark this answers is not in the document]
Comment (sudarshan0212) [reply Ginger/Ginger/UserControlsLib/MultiSelectComboBox.xaml:1]: UCMultiSelectCombobox was old implementation, will remove it.

File: Ginger/GingerCoreCommon/VariablesLib/InputVariableRule.cs [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+                if (value != mOperationValueList)
+                {
+                    mOperationValueList = value;
+                    OnPropertyChanged(nameof(OperationValueList)); 
+                }
+            }
+        }
+
+        //private Dictionary<string, object> mOperationValueList;
+        //[IsSerializedForLocalRepository]
```
[the remark this answers is not in the document]
Comment (sudarshan0212) [reply Ginger/GingerCoreCommon/VariablesLib/InputVariableRule.cs]: removed commented code in all places.

File: Ginger/GingerCoreNET/InputVariableRule/ProcessInputVariableRule.cs [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        public void GetVariablesByRules(ObservableList<VariableBase> variables)
+        {
+            removedbfInputVariables = new ObservableList<VariableBase>();
+            foreach (InputVariableRule variableRule in mBusinessFlow.InputVariableRules)
+            {
+                if (variableRule.Active)
+                {
+                    VariableBase sourceVariable = variables.Where(x => x.Guid == variableRule.SourceVariableGuid).FirstOrDefault();
+                    VariableBase targetVariable = variables.Where(x => x.Guid == variableRule.TargetVariableGuid).FirstOrDefault();
+                    //if (targetVariable == null)
```
[the remark this answers is not in the document]
Comment (sudarshan0212) [reply Ginger/GingerCoreNET/InputVariableRule/ProcessInputVariableRule.cs]: added check if source & target variable is null.

File: Ginger/GingerCoreNET/InputVariableRule/ProcessInputVariableRule.cs [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        public ProcessInputVariableRule(BusinessFlow businessFlow)
+        {
+            mBusinessFlow = businessFlow;
+        }
+
+        public void GetVariablesByRules(ObservableList<VariableBase> variables)
+        {
+            removedbfInputVariables = new ObservableList<VariableBase>();
+            foreach (InputVariableRule variableRule in mBusinessFlow.InputVariableRules)
+            {
```
[the remark this answers is not in the document]
Comment (sudarshan0212) [reply Ginger/GingerCoreNET/InputVariableRule/ProcessInputVariableRule.cs:33]: added try-catch block.

**Review by MeniKadosh1 — DISMISSED** [review]

**Review by MeniKadosh1 — APPROVED** [review]

**Pull request merged**